common/dynamicconfig/cachedvalue.go

```diff
@@ -0,0 +1,59 @@
+package dynamicconfig
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"go.temporal.io/server/common/log/tag"
+)
+
+// GlobalCachedTypedValue holds a cached value of a GlobalTypedSetting after applying a supplied convert function.
+// This type should be deleted after caching is added to GlobalTypedSettingWithConverter.
+type GlobalCachedTypedValue[T any] struct {
+	ptr    *atomic.Pointer[T]
```

Comment:
If GlobalCachedTypedValue is used as a pointer, which it is in this PR, then this shouldn't be a pointer here. We only need one level of indirection